### ISSUE TITLE
Sometimes the Manifest does not contain any OSGi meta data.

### ISSUE THREAD

**Issue by cschneider** [bndtools/bnd#1551]
I realized that when doing just mvn install I sometimes end up with a Manifest that does not have any OSGi meta data. I assume the this Manifest is created by the plain eclipse or maven build.

So there is either a timing issue or the bnd-maven-plugin does not change the Manifest if it is already created.

**Comment by cschneider** [comment]
I found one case where the behaviour of having the plain non OSGi Manifest can be reproduced easily. If I just run mvn clean then I always have a plain Manifest in target/classes/META-INF. Interestingly if I then do mvn install it is overwritten by the OSGi one. So I think the only explanation for the plain Manifest in the jar might be a timing issue between maven and incremental build.

**Comment by bjhargrave** [comment]
You don't mention M2E anywhere but you do mention incremental build. What is the scenario? Is this just command line mvn building? M2E only? Or some combination? Can you make a small github project which can reproduce the issue?

**Comment by cschneider** [comment]
The strange Manifest on mvn clean happens in pure maven but I am not sure if it is relevant.

Otherwise I only saw the problem when I also had the project open in Eclipse with m2e.

**Comment by bjhargrave** [comment]
mvn clean will just clean output, so I would not expect the bnd-maven-plugin to run or even the maven-jar-plugin.

With M2E, the bnd-maven-plugin attempts to handle incremental builds by only processing the target/classes folder if an input has changed:

https://github.com/bndtools/bnd/blob/master/maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java#L167-L179
https://github.com/bndtools/bnd/blob/master/maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/BndMavenPlugin.java#L205

so the issue is likely related to that. Run maven in --debug mode and see what is logged from the bnd-maven-plugin when this occurs.

**Comment by bjhargrave** [comment]
It is possible that the manifest file existence test need to be expanded to also peek in the file to see if there is any OSGi metadata there.

**Comment by cschneider** [comment]
I just tested mvn clean completely outside eclipse. In this case the whole target dir was not there. So it is not a pure maven issue. When I run mvn clean from m2e using run as mvn clean then the non OSGi Manfest is created. So I guess that after eclipse sees the target folder go away it starts (part of) a build.

**Comment by cschneider** [comment]
At the moment I am not able to reproduce the issue with the wrong Manifest in any other case then mvn clean anymore.

It might have to do with the way I set up my pom now:
http://apaste.info/1bV

Btw. I also have some good news. I did some tests how bnd-maven-plugin behhaves on incremental changes and I think it nicely creates and exploded bundle in target/classes. So I am quite positive that we can use this location to install a bundle from during a run/debug.
